packages/E2ETest/app/ImageTestPage.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {StyleSheet, View, Image, Button, requireNativeComponent} from 'react-native'
+import React, { useState } from 'react';
+import { TREE_DUMP_RESULT, SHOW_IMAGE_BORDER, IMAGE_CONTAINER } from './Consts';
+const TreeDumpControl = requireNativeComponent('TreeDumpControl');
+
+const styles = StyleSheet.create({
+  container: {
+    height:300,
+    width:500,
+    backgroundColor: 'yellow',
+    alignItems:'center',
+  },
+  containerWithBorder: {
+    height:300,
+    width:500,
+    borderRadius: 10.0,
+    borderWidth:10,
+    borderColor: '#00ff0055',
+    backgroundColor: 'yellow',
+    alignItems:'center',
+  },
+  imageWithBorder: {
+    height: '100%',
+    width: '100%',
+    borderRadius: 10.0,
+    borderWidth:10,
+    borderColor: '#0000ff55',
+    backgroundColor: 'red',
+  },
+  image: {
+    height: '100%',
+    width: '100%',
+    backgroundColor: 'red',
+  },  
+  treeDumpControl: {
+    height: 150,
+    width: 500,
+    margin: 10,
+  },
+});
+
+export function ImageTestPage() {
+  const [imageWithBorder, setImageBorder] = useState(false);
+  const [clickCount, setClickCount] = useState(0);
+  const onPressBorder = () => {
+    var previousImageBorderState = imageWithBorder;
+    setImageBorder(!previousImageBorderState);
+    var previousClickCount = clickCount;
+    setClickCount(previousClickCount+1);
+ }
+  return(
+  <View>
+    <View testID={IMAGE_CONTAINER} style={imageWithBorder?styles.containerWithBorder:styles.container}>
+      <Image
+        style={imageWithBorder?styles.imageWithBorder:styles.image}
+        resizeMode={'center'}
+        source={{uri: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADMAAAAzCAYAAAA6oTAqAAAAEXRFWHRTb2Z0d2FyZQBwbmdjcnVzaEB1SfMAAABQSURBVGje7dSxCQBACARB+2/ab8BEeQNhFi6WSYzYLYudDQYGBgYGBgYGBgYGBgYGBgZmcvDqYGBgmhivGQYGBgYGBgYGBgYGBgYGBgbmQw+P/eMrC5UTVAAAAABJRU5ErkJggg=='}}        
+      />      
+    </View >
+    <Button title= {imageWithBorder?"Hide Border":"Show Border"} 
+          onPress={onPressBorder} 
+          testID={SHOW_IMAGE_BORDER}/> 
+    <TreeDumpControl style={styles.treeDumpControl} dumpID={imageWithBorder?'ImageWithBorder':'ImageWithoutBorder'+clickCount} uiaID={IMAGE_CONTAINER} testID={TREE_DUMP_RESULT} />
```

Comment:
>'ImageWithoutBorder'+clickCount [](start = 94, length = 31)

ImageWithoutBorder0.txt and ImageWithoutBorder2.txt is not readable, also you created a hard binding between ImageTestPage and Image.spec.ts.  If you refactor the test case like change the order or add a new click, it doesn't work anymore.

Reply:
I am going to refactor the dumpID to "ImageWithoutBorder" and "ImageWithoutBorder-Subsequent", so it would work for click count > 2. The dump file is different if you initially don't have a border, or had a border then removed the border.

---
In reply to: [358381794](https://github.com/microsoft/react-native-windows/pull/3754#discussion_r358381794) [](ancestors = 358381794)